src/DataTypes/Serializations/SerializationBool.cpp

```diff
@@ -0,0 +1,138 @@
+
+#include <Columns/ColumnsNumber.h>
+#include <Common/Exception.h>
+#include <IO/WriteBuffer.h>
+#include <IO/ReadBuffer.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int CANNOT_PARSE_DOMAIN_VALUE_FROM_STRING;
+    extern const int ILLEGAL_COLUMN;
+}
+
+SerializationBool::SerializationBool(const SerializationPtr &nested_)
+        : SerializationCustomSimpleText(nested_)
+{
+}
+
+void SerializationBool::serializeText(const IColumn &column, size_t row_num, WriteBuffer &ostr, const FormatSettings &) const
+{
+    const auto *col = checkAndGetColumn<ColumnUInt8>(&column);
+    if (!col)
+        throw Exception("Bool type can only serialize columns of type UInt8." + column.getName(),
+                        ErrorCodes::ILLEGAL_COLUMN);
+
+    if (col->getData()[row_num])
+        ostr.write(str_true, sizeof(str_true) - 1);
+    else
+        ostr.write(str_false, sizeof(str_false) - 1);
+}
+
+void SerializationBool::deserializeText(IColumn &column, ReadBuffer &istr, const FormatSettings &) const
+{
+    ColumnUInt8 *col = typeid_cast<ColumnUInt8 *>(&column);
+    if (!col)
+    {
+        throw Exception("Bool type can only deserialize columns of type UInt8." + column.getName(),
+                        ErrorCodes::ILLEGAL_COLUMN);
+    }
+
+    if (!istr.eof())
+    {
+        bool value = false;
+
+        if (*istr.position() == 't' || *istr.position() == 'f')
+            readBoolTextWord(value, istr);
+        else if (*istr.position() == '1' || *istr.position() == '0')
+            readBoolText(value, istr);
+        else
+            throw Exception("Invalid boolean value, should be true, false, 1, or 0.",
+                            ErrorCodes::CANNOT_PARSE_DOMAIN_VALUE_FROM_STRING);
+        col->insert(value);
+    }
+    else
+        throw Exception("Expected boolean value but get EOF.", ErrorCodes::CANNOT_PARSE_DOMAIN_VALUE_FROM_STRING);
+}
+
+void SerializationBool::serializeTextEscaped(const IColumn & column, size_t row_num, WriteBuffer & ostr, const FormatSettings & settings) const
+{
+    const auto *col = checkAndGetColumn<ColumnUInt8>(&column);
+    if (!col)
+        throw Exception("Bool type can only serialize columns of type UInt8." + column.getName(),
+                        ErrorCodes::ILLEGAL_COLUMN);
+    if (col->getData()[row_num])
+    {
+        writeString(settings.bool_true_representation, ostr);
+    }
+    else
+    {
+        writeString(settings.bool_false_representation, ostr);
+    }
+}
+
+void SerializationBool::deserializeTextEscaped(IColumn & column, ReadBuffer & istr, const FormatSettings & settings) const
+{
+    ColumnUInt8 * col = typeid_cast<ColumnUInt8 *>(&column);
+    if (!col)
+    {
+        throw Exception("Bool type can only deserialize columns of type UInt8." + column.getName(), ErrorCodes::ILLEGAL_COLUMN);
+    }
+
+    if (!istr.eof())
+    {
+        String input;
```

Comment:
I prefer if we will be permissive while deserializing.

So, we try bool_true_representation/bool_false_representation first, and if it does not match - try True/False/T/F/Y/N/Yes/No/On/Off, all case insensitive.

Another idea - make the settings for representations as arrays?
But it's also interesting to keep minimal overhead for parsing (don't just compare in a loop).

Reply:
@Avogar what do you think?

Reply:
> So, we try bool_true_representation/bool_false_representation first, and if it does not match - try True/False/T/F/Y/N/Yes/No/On/Off, all case insensitive.

Good idea.

> Another idea - make the settings for representations as arrays?
But it's also interesting to keep minimal overhead for parsing (don't just compare in a loop).

What about making the setting as regex?

Reply:
what about keep bool_true_representation/bool_false_representation and another way both , and add a setting to control it . for those people who don't want higher cost , he can chose strict mode. otherwise can chose more permissive way.

Reply:
> what about keep bool_true_representation/bool_false_representation and another way both , and add a setting to control it . for those people who don't want higher cost , he can chose strict mode. otherwise can chose more permissive way.

Sounds good, @alexey-milovidov what do you think?

Reply:
Making it as regexp is not an option as it will slow down processing.
Initially I thought to hardcode True/False/T/F/Y/N/Yes/No/On/Off, as we can write a tight implementation to recognize them essentially for free...

Reply:
Ok, then let's implement your suggestion when bool_true_representation/bool_false_representation does not match we try True/False/T/F/Y/N/Yes/No/On/Off case insensitive.

Reply:
If you like, we can merge this PR and then implement the adjustments...